pkg/kubectl/cmd/util/editor/doc.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package editor is used to complete edit options and open an editor
+// to apply changes to resources/objects for running edit cli command.
+// Example for usage for cli command as follows:
+//
+// f, _, _, _ := testing.NewAPIFactory()
```

Comment:
Can you think of anyway we could make that into a compilable example, as you did in the other PR (or described [here](https://blog.golang.org/examples))?

Reply:
Do you mean that examples are tests? If so, I will add some testcases

Reply:
Examples are indeed executed as tests, but they are also printed nicely in the documentation. It helps to keep the example up-to-date with the code and make sure they compile and run successfully. The main different with a test, is that their function signature is different. Read the blog post below.

Your example is pretty big, so I think it would make sense to follow the "Larger Example" section from the (golang blogpost)[https://blog.golang.org/examples].

Reply:
Thanks for your suggestion. 
I should add Example() into editor_test.go. Right?

Reply:
If you write a "Larger Example", it needs to be in its own file. `example_test.go` maybe?

> To achieve this we can use a "whole file example." A whole file example is a file that ends in _test.go and contains exactly one example function, no test or benchmark functions, and at least one other package-level declaration.

Reply:
@apelisse I find there are many work to add Example() for editor. Editor is interactive between kubectl and api-server.
Could I add a ExampleEditOption() with fake httpserver reference the TestEdit()(  (https://github.com/kubernetes/kubernetes/blob/210626577b17ab014589e983b50b7cb821181a6c/pkg/kubectl/cmd/edit_test.go#L77)?

I think Example() is a successful testcase.